tangelo/toolboxes/ansatz_generator/hea.py

```diff
@@ -44,18 +44,21 @@ class HEA(Ansatz):
         n_qubits (int) : The number of qubits in the ansatz.
             Default, None.
         n_electrons (int) : Self-explanatory.
-        reference_state (str): "HF": Hartree-Fock reference state. "zero": for
-            no reference state. Default: "HF".
+        reference_state (str, Circuit): "HF": Hartree-Fock reference state. "zero": for
+            no reference state. Can also be a Circuit object, in which case a copy of
+            circuit with variational parameters fixed is used. Default: "HF".
         """
 
     def __init__(self, molecule=None, mapping="jw", up_then_down=False,
                  n_layers=2, rot_type="euler", n_qubits=None, n_electrons=None,
                  spin=None, reference_state="HF"):
 
-        if not (bool(molecule) ^ (bool(n_qubits) and (bool(n_electrons) | (reference_state == "zero")))):
+        if not (bool(molecule) ^ (bool(n_qubits) and
```

Comment:
This is starting to be complicated to parse, even for developers. 

Would there be a way to split this statement into maybe two simpler ones with more targetted error messages ?
It is ok if the code is less compact, I think here the challenge is about readability.

If there's no convincing solution, this is ok and we can move on

Reply:
Fixed in [commit 37a9497](https://github.com/goodchemistryco/Tangelo/pull/398/commits/37a94975fdc86946aeb7c4a7e5a0f9a7555aa9b7) and [commit 76af054](https://github.com/goodchemistryco/Tangelo/pull/398/commits/76af054e2bb737a8e51c30eb4195d309d097df72). Let me know if you have any suggestions for further improving the clarity of the logic and error messages.